### PR TITLE
Use Cypher aggregation instead of cypher projection in graph.construct

### PR DESCRIPTION
**Pull request by FlorentinD**
- Use cypher aggregation for community users
- Handle overlapping property names in rel and nodes df (WIP)

Thank you for your contribution to the Graph Data Science Client project.

Before submitting this PR, please read [Contributing to the Neo4j Ecosystem](https://github.com/neo4j/graph-data-science-client/blob/main/CONTRIBUTING.md). 

Make sure:
- [x] You signed the [Neo4j CLA](https://neo4j.com/developer/cla/#sign-cla) (Contributor License Agreement) so that we are allowed to ship your code in our library
- [x] Your contribution is covered by tests

### CODE REVIEW

**Comment by FlorentinD** [comment]
[[Directed] Use cypher aggregation in the python client](https://trello.com/c/3ibWqlku/7326-directed-use-cypher-aggregation-in-the-python-client)

File: graphdatascience/query_runner/cypher_graph_constructor.py [hunk 1]
```diff
@@ -73,51 +71,244 @@ def _should_warn_about_arrow_missing(self) -> bool:
...
+
+            nodes_config = self.nodes_config(combined_cols, graph_schema.nodes.all)
+            rels_config = self.rels_config(combined_cols, graph_schema.relationships.all)
+
+            property_clauses: List[str] = []
+
+            all_prop_cols = sorted(list(graph_schema.relationships.properties) + list(graph_schema.nodes.properties))
+
+            for prop_col in all_prop_cols:
+                property_clauses.append(self.check_value_clause(combined_cols, prop_col))
```
Comment (knutwalker) [inline graphdatascience/query_runner/cypher_graph_constructor.py:114]: maybe with a list-comprehension?

```suggestion
            all_prop_cols = sorted(list(graph_schema.relationships.properties) + list(graph_schema.nodes.properties))

            property_clauses = [self.check_value_clause(combined_cols, prop_col) for prop_col in all_prop_cols]
```

File: graphdatascience/query_runner/cypher_graph_constructor.py [hunk 2]
```diff
@@ -73,51 +71,244 @@ def _should_warn_about_arrow_missing(self) -> bool:
...
+            nodes_config_fields: List[str] = []
+
+            if "labels" in node_cols:
+                nodes_config_fields.append(f"sourceNodeLabels: data[{combined_cols.index('sourceNodeLabels')}]")
+
+            property_columns: List[str] = list(set(node_cols) - {"nodeId", "labels"})
+            property_columns.sort()
+
+            # as we first list all nodes at the top of the df, we dont need to lookup properties for the target node
+            if len(property_columns) > 0:
```
Comment (knutwalker) [inline graphdatascience/query_runner/cypher_graph_constructor.py:216]: I believe some linter would suggest to use 

```suggestion
            if property_columns:
```

as empty lists/sets/anything are falsey

**Review by knutwalker — APPROVED** [review]
noice 👍

File: graphdatascience/query_runner/cypher_graph_constructor.py [hunk 3]
```diff
@@ -32,28 +44,14 @@ def run(self, node_dfs: List[DataFrame], relationship_dfs: List[DataFrame]) -> N
...
-                "read_concurrency": self._concurrency,
-                "nodes": nodes,
-                "relationships": relationships,
-            },
-        )
+        node_df = node_dfs[0]
+        rel_df = relationship_dfs[0]
+
+        # Cypher aggregation supports concurrency since 2.3.0
+        if self._server_version >= ServerVersion(2, 3, 0):
```
[the remark this answers is not in the document]
Comment (FlorentinD) [reply graphdatascience/query_runner/cypher_graph_constructor.py:51]: Probably, but I would like to stick to just use the server version for now. Then its a smaller change for the user when they only update the client.

**Pull request merged**